internal/config/validator_test.go

```diff
@@ -28,3 +29,255 @@ func TestValidator(t *testing.T) {
 		t.Error("The validator should not fail for non existing bfd profile")
 	}
 }
+func TestExistCommunity(t *testing.T) {
+	tests := []struct {
+		community   string
+		communities []metallbv1beta1.Community
+		expect      bool
+	}{
+		{
+			community: "abcd",
+			communities: []metallbv1beta1.Community{
+				{
+					Spec: metallbv1beta1.CommunitySpec{
+						Communities: []metallbv1beta1.CommunityAlias{
+							{
+								Name:  "myFirstCommunityAlias",
+								Value: "65000:100",
+							},
+							{
+								Name:  "mySecondCommunityAlias",
+								Value: "65000:200",
+							},
+						},
+					},
+				},
+			},
+			expect: false,
+		},
+		{
+			community: "myFirstCommunityAlias",
+			communities: []metallbv1beta1.Community{
+				{
+					Spec: metallbv1beta1.CommunitySpec{
+						Communities: []metallbv1beta1.CommunityAlias{
+							{
+								Name:  "myFirstCommunityAlias",
+								Value: "65000:100",
+							},
+							{
+								Name:  "mySecondCommunityAlias",
+								Value: "65000:200",
+							},
+						},
+					},
+				},
+			},
+			expect: true,
+		},
+	}
+	for _, test := range tests {
+		result := existCommunity(test.community, test.communities)
+		if result != test.expect {
+			t.Errorf("%s: expect is %v, but result is %v", test.community, test.expect, result)
+		}
+	}
+}
+func TestResetTransientErrorsFields(t *testing.T) {
+	tests := []struct {
+		clusterResources       ClusterResources
```

Comment:
add a "desc" field

Reply:
In the struct? Why? Maybe I don't understand what you mean.
Can you give me an example?

Reply:
look for `test.desc` in the code, together with t.Run it gives each testcase a name (description) and it makes it easier to understand which scenario failed